src/core/ext/filters/client_channel/xds/xds_bootstrap.cc

```diff
@@ -131,19 +155,29 @@ grpc_error* XdsBootstrap::ParseXdsServer(grpc_json* json) {
             "duplicate \"channel_creds\" field"));
       }
       seen_channel_creds = true;
-      grpc_error* parse_error = ParseChannelCredsArray(child);
+      grpc_error* parse_error = ParseChannelCredsArray(child, &server);
       if (parse_error != GRPC_ERROR_NONE) error_list.push_back(parse_error);
     }
   }
-  if (server_uri_ == nullptr) {
+  if (server.server_uri == nullptr) {
     error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
         "\"server_uri\" field not present"));
   }
-  return GRPC_ERROR_CREATE_FROM_VECTOR("errors parsing \"xds_server\" object",
-                                       &error_list);
+  // Can't use GRPC_ERROR_CREATE_FROM_VECTOR() here, because the error
+  // string is not static in this case.
+  if (error_list.empty()) return GRPC_ERROR_NONE;
+  char* msg;
+  gpr_asprintf(&msg, "errors parsing index %" PRIuPTR, idx);
+  grpc_error* error = GRPC_ERROR_CREATE_FROM_COPIED_STRING(msg);
+  gpr_free(msg);
+  for (size_t i = 0; i < error_list.size(); ++i) {
+    error = grpc_error_add_child(error, error_list[i]);
+  }
+  return error;
```

Comment:
If we have an error, probably we should remove the server from servers_ before returning?

Reply:
As above, it doesn't matter, because the config parsing will fail anyway.